bot/exts/moderation/incidents.py

```diff
@@ -402,10 +492,104 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
+        link), this message link embeds are then sent into the channel.
+
+        Also passes the message into `add_signals` if the message is a incident.
+        """
         if is_incident(message):
             await add_signals(message)
 
+            webhook_embed_list = await extract_message_links(message, self.bot)
+            if webhook_embed_list:
+                await self.send_webhooks(webhook_embed_list, message, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_message_edit(self, msg_before: discord.Message, msg_after: discord.Message) -> None:
+        """
+        Pass `msg_after` to `extract_message_links` and edit `msg_before` webhook msg.
+
+        Deletes cache value (`message_link_embeds_cache`) of `msg_before` if it exists and removes the
+        webhook message for that particular link from the channel.
+
+        If the message edit (`msg_after`) is a incident then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link), this message link embeds are then sent into the channel.
+
+        The edited message is also passed into `add_signals` if it is a incident message.
+        """
+        if is_incident(msg_after):
+            webhook_embed_list = await extract_message_links(msg_after, self.bot)
+            webhook_msg_id = await self.message_link_embeds_cache.get(msg_before.id)
+
+            if webhook_msg_id:
+                await self.incidents_webhook.edit_message(
+                    message_id=webhook_msg_id,
+                    embeds=[x for x in webhook_embed_list if x is not None],
+                )
+                return
+
+            await self.send_webhooks(webhook_embed_list, msg_after, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_message_delete(self, message: discord.Message) -> None:
+        """
+        Delete message link embeds for `message`.
+
+        Search through the cache for message, if found delete it from cache and channel.
+        """
+        if is_incident(message):
+            await self.delete_msg_link_embed(message)
+
+    async def send_webhooks(
+            self,
+            webhook_embed_list: t.List,
+            message: discord.Message,
+            webhook: discord.Webhook,
+    ) -> t.Optional[int]:
+        """
+        Send Message Link Embeds to #incidents channel.
+
+        Uses the `webhook` passed in as a parameter to send
+        the embeds in the `webhook_embed_list` parameter.
+
+        After sending each webhook it maps the `message.id`
+        to the `webhook_msg_ids` IDs in the async redis-cache.
+        """
+        try:
+            webhook_msg = await webhook.send(
+                embeds=[x for x in webhook_embed_list if x is not None],
+                username=sub_clyde(message.author.name),
+                avatar_url=message.author.avatar_url,
```

Comment:
This is an interesting idea, but I'm not sure if it fits here. Maybe just the Python logo and the name "Incident Context"? or maybe some info symbol for the avatar.

Reply:
You know what scratch that, it can help identify who sent what.